src/activity_actor.cpp

```diff
@@ -374,6 +374,41 @@ std::unique_ptr<activity_actor> migration_cancel_activity_actor::deserialize( Js
     return migration_cancel_activity_actor().clone();
 }
 
+void purify_water_activity_actor::start( player_activity &act, Character & )
+{
+    act.moves_total = moves;
+    act.moves_left = moves;
+}
+
+void purify_water_activity_actor::finish( player_activity &act, Character & )
+{
+    liquid.get_item()->convert( "water_clean" ).poison = 0;
```

Comment:
Looking good! Last suggestion then I think this is good to go.  It's safer to check if the item location is still valid before accessing the item.
```suggestion
    if( liquid ) {
        liquid->convert( "water_clean" ).poison = 0;
    } else {
        debugmsg( "Lost target item of ACT_PURIFY_WATER" );
    }
```

Reply:
Thanks for your help!